src/test/java/com/shika/testCases/ACourseCategoryTest.java

```diff
@@ -7,26 +7,60 @@
 import org.testng.annotations.Test;
 
 import com.shika.dataproviders.DataProviderClass;
+import com.shika.pages.ACourseCategoryPage;
 import com.shika.pages.ACreateCourseCategoryPage;
 import com.shika.testbase.TestBase;
 
 public class ACourseCategoryTest extends TestBase {
 
-	@Test(dataProvider = "createCategory", dataProviderClass = DataProviderClass.class)
+	@Test(dataProvider = "categoryList", dataProviderClass = DataProviderClass.class)
 
-	public void createCourseCategory(String testDesc, String name, String expMessage, Method method) {
+	public void courseCategory(String testDesc, String name, Method method) throws InterruptedException {
 		
 		logger = extent.createTest(method.getName().toString());
-
-		ACreateCourseCategoryPage categoryPage = new ACreateCourseCategoryPage(driver);
+		
+		ACourseCategoryPage categoryPage = new ACourseCategoryPage(driver);
 
 		driver.get(properties.getProperty("url")+"administrator/index.php?option=com_categories&extension=com_tjlms");
 		
 		if (testDesc.equals("valid")) {
 
-			categoryPage.createCategoryAs(name);
-
-		}
+			// Unpublish single category
+			categoryPage.searchCategory(name);
+			Thread.sleep(2000);
+			categoryPage.unpublishSigleCategory();
+			
+			
+			//Publish single category
+			categoryPage.clearSearch();
```

Comment:
We can reduce the lines of code here. You can re-write the code as -

`categoryPage.clearSearch().searchCategory(name).publishSingleCategory();
`